implicit/als.py

```diff
@@ -81,12 +81,12 @@ def fit(self, item_users):
                                            self.regularization, num_threads=self.num_threads)
                 log.debug("loss at iteration %i is %s", iteration, loss)
 
-    def recommend(self, userid, user_items, N=10):
+    def recommend(self, userid, user_items, N=10, filter_items=[]):
```

Comment:
Having a list or other structure as a default value can cause problems in Python. See this post for details: http://python-guide-pt-br.readthedocs.io/en/latest/writing/gotchas/#mutable-default-arguments

I would prefer passing the default as ```def recommend(self, userid, user_items, N=10, filter_items=None)``` and then go
```
liked = set(user_items[userid].indices)
if filter_items:
    liked.update(filter_items)
```
This also means that its possible to pass in any iterable (list/generator/set etc) instead of just a list